stackslib/src/burnchains/bitcoin/indexer.rs

```diff
@@ -1000,9 +1000,10 @@ impl BurnchainIndexer for BitcoinIndexer {
             false,
         )?;
         let first_block_height = self.get_first_block_height();
-        let first_header = spv_client
-            .read_block_header(first_block_height)?
-            .expect("BUG: no first block header hash");
+        let first_header = match spv_client.read_block_header(first_block_height) {
+            Ok(Some(res)) => res,
+            _ => return Err(burnchain_error::MissingHeaders),
+        };
```

Comment:
`let-else` is simpler here:

```suggestion
        let Ok(Some(first_header)) = spv_client.read_block_header(first_block_height) else {
            return Err(burnchain_error::MissingHeaders);
        };
```